app/src/main/java/com/example/onsalestore/activities/ItemDetailActivity.java

```diff
@@ -11,37 +12,43 @@
 import android.widget.Toast;
 
 import androidx.appcompat.widget.Toolbar;
+import androidx.coordinatorlayout.widget.CoordinatorLayout;
 
 import com.bumptech.glide.Glide;
 import com.example.onsale.R;
 import com.example.onsalestore.objects.ClosetItem;
 import com.example.onsalestore.objects.ClothingItem;
+import com.google.android.material.snackbar.Snackbar;
+import com.parse.Parse;
 import com.parse.ParseException;
+import com.parse.ParseObject;
 import com.parse.ParseUser;
 import com.parse.SaveCallback;
 
 import org.parceler.Parcels;
 
 public class ItemDetailActivity extends AppCompatActivity {
 
-    TextView itemDetailName, itemDetailPrice, itemDetailSource, itemDetailUrl;
-    ImageView itemDetailImage;
-    Button btnAddToCloset;
+    private TextView itemDetailName;
+    private TextView itemDetailPrice;
+    private TextView itemDetailSource;
+    private TextView itemDetailUrl;
+    private ImageView itemDetailImage;
+    private Button btnAddToCloset;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_item_detail);
-
+        ClothingItem itemDetail = Parcels.unwrap(getIntent().getParcelableExtra("EXTRA_ITEM"));
+        setTitle("" + itemDetail.getItemName());
```

Comment:
`getItemName` already returns a string, no need to concatenate with an empty one.
```suggestion
        setTitle(itemDetail.getItemName());
```